### ISSUE TITLE
cpuinfo still crashes on Apple Silicon due to checking cache size

### ISSUE THREAD

**Issue by t-mitchell** [BJTor/CPUInfo#4]
Hi,

On an Apple Silicon M2 Pro and R2023a, cpuinfo still crashes because it still attempt to get the cache size.  Specifically, line 111 is executed, which results in this error:

```Unrecognized field name "cache".

Error in cpuinfo>cpuInfoMac (line 111)
    cacheBytes = str2double(machdep.cache.size)*1024; % convert from kB

Error in cpuinfo (line 20)
        info = cpuInfoMac();
```

The real problem appears to be the comparison in line 106 because `computer` is actually set to `MACI64`, not `MACA64` as expected.  Presumably this is because Matlab R2023a is still a program compiled for Intel architecture and is being translated using Rosetta 2.

**Comment by BJTor** [comment]
Thanks for tracking down the issue - I had only tried this using native builds. Working on it now.

**Comment by BJTor** [comment]
Should be fixed in latest version. I'll do a new release. I have checked this on an Apple M2 Macbook Air using both native build and via Rosetta and both now work without error and give the same output.

**Comment by t-mitchell** [comment]
Thanks for the quick turn around.  I just tested on R2023a Update 4 using a MacBook Pro with an M1 Pro (10 CPU cores, 16 GPU cores) and a Mac Mini with an M2 Pro (12 CPU cores, 19 GPU cores) and both worked fine!  I also tried to test with Update 5 but when I asked Matlab to install it, it gave me a 401 error (!?) in a pop-up window with a link to a rather general webpage that wasn't helpful.  I don't have the prerelease of R2023b installed to test the native Apple Silicon build, but if you'd like me to test it, I suppose I can install the prerelease.  I've been fine using R2023a until R2023b is officially released.

By the way, thanks for making cpuinfo.m available in the first place.  It's a super useful utility, and I think should be included with MATLAB!
